osbs/core.py

```diff
@@ -705,6 +705,54 @@ def get_image_stream_tag(self, tag_id):
         check_response(response)
         return response
 
+    def put_image_stream_tag(self, tag_id, tag):
+        url = self._build_url("imagestreamtags/%s" % tag_id)
+        response = self._put(url, data=tag,
+                             headers={"Content-Type": "application/json"})
+        check_response(response)
+        return response
+
+    def ensure_image_stream_tag(self, stream, tag_id, tag_template, scheduled=False):
```

Comment:
Agree. I removed the requirement to convert to Json. Instead, this is now done internally right before calling the actual REST API. This should make usage easier.